tests/frontend/helper.js

```diff
@@ -112,9 +112,20 @@ var helper = {};
       helper.clearSessionCookies();
     }
 
+    var rootPath = '/p/';
+    var padNameRadnomString = 4;
+    var depth = Math.floor((Math.random() * 5) + 1);
+    for(var i=1; i <= depth; i++) {
+      var padPrefix = helper.randomString(3);
+      rootPath += padPrefix + '/';
+    }
+
     if(!padName)
-      padName = "FRONTEND_TEST_" + helper.randomString(20);
-    $iframe = $("<iframe src='/p/" + padName + (encodedParams || '') + "'></iframe>");
+      padName = rootPath + "FRONTEND_TEST_" + helper.randomString(padNameRadnomString);
+
+    padName = padName.toString().indexOf('/p/') !== 0 ? rootPath + padName : padName;
+
+    $iframe = $("<iframe src='" + padName + (encodedParams || '') + "'></iframe>");
```

Comment:
All test cases are now run in a nested document with a random depth path.
@JohnMcLear